algoliasearch/src/main/java/com/algolia/search/saas/Index.java

```diff
@@ -669,20 +680,53 @@ protected JSONObject getObject(String objectID, List<String> attributesToRetriev
      * @throws AlgoliaException
      */
     protected JSONObject getObjects(List<String> objectIDs) throws AlgoliaException {
+        return getObjects(objectIDs, null);
+    }
+
+    /**
+     * Get several objects from this index
+     *
+     * @param objectIDs            the array of unique identifier of objects to retrieve
+     * @param attributesToRetrieve contains the list of attributes to retrieve.
+     * @throws AlgoliaException
+     */
+    protected JSONObject getObjects(List<String> objectIDs, List<String> attributesToRetrieve) throws AlgoliaException {
         try {
             JSONArray requests = new JSONArray();
             for (String id : objectIDs) {
                 JSONObject request = new JSONObject();
                 request.put("indexName", this.indexName);
                 request.put("objectID", id);
+                request.put("attributesToRetrieve", encodeAttributes(attributesToRetrieve, false));
                 requests.put(request);
             }
             JSONObject body = new JSONObject();
             body.put("requests", requests);
             return client.postRequest("/1/indexes/*/objects", body.toString(), true);
         } catch (JSONException e) {
             throw new AlgoliaException(e.getMessage());
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Nullable
+    private String encodeAttributes(List<String> attributesToRetrieve, boolean forURL) throws UnsupportedEncodingException {
+        if (attributesToRetrieve == null) {
+            return null;
+        }
+
+        StringBuilder params = new StringBuilder();
+        if (forURL) {
+            params.append("?attributes=");
```

Comment:
`attributes` is deprecated. It's `attributesToRetrieve` now. Instead of manually mingling with the attributes, I would create a `Query`, call `setAttributesToRetrieve()` then `build()` and boom! here you go: simpler, more reliable.